opennlp-tools/src/main/java/opennlp/tools/commons/Trainer.java

```diff
@@ -17,8 +17,36 @@
 
 package opennlp.tools.commons;
 
+import java.util.Map;
+
+import opennlp.tools.ml.AbstractTrainer;
+import opennlp.tools.ml.EventModelSequenceTrainer;
+import opennlp.tools.util.TrainingParameters;
+
 /**
  * Represents a common base for training implementations.
  */
 public interface Trainer {
+
+  /**
+   * Initializes a {@link AbstractTrainer} via {@link TrainingParameters} and
+   * a {@link Map report map}.
+   *
+   * @param trainParams The {@link Map} that maps training parameters on a key-value basis.
+   * @param reportMap The {@link Map} instance used as report map.
+   *                  
+   * @deprecated Use {@link #init(TrainingParameters, Map)} instead.
+   */
+  @Deprecated
+  void init(Map<String, Object> trainParams, Map<String, String> reportMap);
```

Comment:
If it is `@Deprecated` (as it was pulled up from some implementing classes), why should we keep it in the interface, i.e. shouldn't we just remove it here and migrate the remaining four usages to the other signature?

We already broke binary compatibility, so imho it would be a good opportunity to get rid of it instead of keeping it in the interface. Wdyt, @mawiesne @jzonthemtn ?

Reply:
Resolved with latest commit. The oldskool method now resides in `AbstractTrainer` not in the `Trainer` interface.